src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -45,12 +47,22 @@ public void storeMessage(Message message) {
     datastore.put(messageEntity);
   }
 
-    /** Stores FoodItem in Datastore. */
+    /** Stores FoodItem in Datastore.
+      * Prevent the same food from being entered into the database more than once
+      */
+
   public void storeFood(FoodItem food) {
-    Entity foodEntity = new Entity("FoodItem", food.getID().toString());
-    foodEntity.setProperty("Name", food.getName());
-    foodEntity.setProperty("CO2", food.getCO2PerYear());
-    datastore.put(foodEntity);
+    Set<String> existingfood = this.getFoodItem();
+    if(existingfood.contains(food.getName())){
```

Comment:
You will want to compare all of the foods names in lowercase (or uppercase, just needs to be consistent) so that `Apple` and `apple` aren't different foods.